### PR TITLE
Wrong parsing of "error" json member

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
In case is array or object.

Also reset color "red" for rejected shares.
Addresses #1065

### CODE REVIEW

**Comment by jean-m-cyr** [comment]
This is odd! No one has approved, yet I see a big green 'merge pull request' button?

**Comment by AndreaLanfranchi** [comment]
Yep ... I noticed.
Nevertheless I never merge without at least one positive review.
Reading documentation probably I've been given write access to repo.

File: libpoolprotocols/stratum/EthStratumClient.cpp [hunk 1]
```diff
@@ -554,11 +574,31 @@ void EthStratumClient::processReponse(Json::Value& responseObject)
...
-			_isSuccess = !responseObject.isMember("error");
+			_isSuccess = !responseObject.isMember("error") || responseObject.get("error", Json::Value::null).isNull();
 
 			if (!_isSuccess) {
-				if (responseObject.isMember("error")) {
-					_errReason = responseObject.get("error", "Unknown error").asString();
+
+				if (!responseObject.get("error", Json::Value::null).isNull()) {
+
+					if (responseObject["error"].isConvertibleTo(Json::ValueType::stringValue)) {
```
Comment (jean-m-cyr) [inline libpoolprotocols/stratum/EthStratumClient.cpp]: This block is very similar, if not identical, to the one above. Perhaps make it into a method.

Comment (AndreaLanfranchi) [reply libpoolprotocols/stratum/EthStratumClient.cpp]: Yep you're right.
Sometimes laziness afflicts me too. :)
Wilco.

**Comment by chfast** [comment]
The `release/0.14` was not a protected branch. I changed that.

**Pull request merged**